pkgs/build-support/fetchgit/default.nix

```diff
@@ -55,12 +55,14 @@ in
 */
 
 assert deepClone -> leaveDotGit;
-assert nonConeMode -> (sparseCheckout != "");
+assert nonConeMode -> (sparseCheckout != []);
 
 if md5 != "" then
   throw "fetchgit does not support md5 anymore, please use sha256"
 else if hash != "" && sha256 != "" then
   throw "Only one of sha256 or hash can be set"
+else if builtins.isString sparseCheckout then
+  throw "Passing a (multi-line) string is not supported anymore, please provide files/patterns as a list of strings"
```

Comment:
Might want to explicitly mention `sparseCheckout` in the error message, since this does not produce a trace by default.

Reply:
Rephrased mentioning `sparseCheckout`.